src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java

```diff
@@ -1,30 +1,113 @@
+
+    @Autowired
+    private AuditServiceImpl auditService;
+
+    private Date fileTimeStamp;
+
+    @Autowired
+    @Qualifier("credscloudStorageAccount")
+    private CloudStorageAccount cloudStorageAccount;
+
     @Override
     public void process(Exchange exchange) {
         log.info("{}:: FileReadProcessor starts::", logComponentName);
-        String blobFilePath = (String) exchange.getProperty(MappingConstants.BLOBPATH);
-        CamelContext context = exchange.getContext();
-        ConsumerTemplate consumer = context.createConsumerTemplate();
+        final String blobFilePath = (String) exchange.getProperty(BLOBPATH);
+        final CamelContext context = exchange.getContext();
+        final ConsumerTemplate consumer = context.createConsumerTemplate();
+        RouteProperties routeProperties = (RouteProperties) exchange.getIn().getHeader(ROUTE_DETAILS);
+        String fileName = routeProperties.getFileName();
+        String schedulerTime = context.getGlobalOption(SCHEDULER_START_TIME);
+
+        //Check Stale OR Not existing file and exit run with proper error message
+        if (getFileStatusInBlobContainer(fileName, schedulerTime).equals(STALE)) {
+            exchange.getMessage().setHeader(IS_FILE_STALE, true);
+            auditService.auditException(exchange.getContext(), String.format(
+                "%s file with timestamp %s not loaded due to file stale error",
+                fileName,
+                DateFormatUtils.format(fileTimeStamp, "yyyy-MM-dd HH:mm:SS")));
+            return;
+        } else if (getFileStatusInBlobContainer(fileName, schedulerTime).equals(NOT_EXISTS)) {
+            auditService.auditException(exchange.getContext(), String.format(
+                "%s file is not exists in container", routeProperties.getFileName()));
+            return;
+        }
+
+        exchange.getMessage().setHeader(IS_FILE_STALE, false);
+        context.getGlobalOptions().put(fileName, NOT_STALE);
         exchange.getMessage().setBody(consumer.receiveBody(blobFilePath, fileReadTimeOut));
     }
+
+    /**
+     * Connects to Blob storage and checks if file timestamp is stale.
+     *
+     * @param fileName for getting timestamp of
+     * @return is File TimeStamp Stale
+     */
+    private BlobStatus getFileStatusInBlobContainer(String fileName, String schedulerTime) {
+        try {
+
+            CloudBlobClient blobClient = cloudStorageAccount.createCloudBlobClient();
+            CloudBlobContainer container =
+                blobClient.getContainerReference(azureBlobConfig.getContainerName());
+
+            CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(fileName);
+
+            if (cloudBlockBlob.exists()) {
+                cloudBlockBlob.downloadAttributes();
+                fileTimeStamp = cloudBlockBlob.getProperties().getCreatedTime();
+                return (isSameDay(fileTimeStamp, new Date(Long.valueOf(schedulerTime)))) ? NEW : STALE;
+            }
+
```

Comment:
i would suggest to check the status of the previous day job as well.. If it is failed then we should process and not mark it stale
Its a edge case but probably we can handle

Reply:
As discussed will create new task and groom it.